File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{

// Gets text from user to initially evaluate.
    string reading = get_string("Text: ");

// Initializes each variable that will be input into our index equation
    int letters = 0;
    int words = 1;
    int sentences = 0;

// Checks for alphabetical characters by using isalpha from the ctype library and adds those to letters.
    for (int i = 0; i < strlen(reading); i++)
    {
        if (isalpha(reading[i]))
        {
            letters++;
        }
    }

// Checks for spaces after each word and adds those to words. Foreseeable problem with compound words that utilize hyphens.
    for (int i = 0; i < strlen(reading); i++)
    {
        if (isspace(reading[i]))
        {
            words++;
        }
    }

// Checks for punctuation and adds up number of total sentences in text.
    for (int i = 0; i < strlen(reading); i++)
    {
        if (reading[i] == '.' || reading[i] == '!' || reading[i] == '?')
        {
            sentences++;
        }
    }

// Turns letters, words, and sentences into floats and determines number of letters/sentences in every 100 words.
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / words * 100;

// Rounds indexing equation for later.
    int level = round(0.0588 * L - 0.296 * S - 15.8);

// Prints out necessary results.
    if (level < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (level > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", level);
    }
}

File: sentimental-readability/readability.py
count_letter = 0
count_word = 1
count_sentence = 0

# First, we must collect input from the user
text = input("Text: ")
text_length = len(text)

# The next portion counts the # of letters in the text
for i in range(text_length):
    if (text[i].isalpha()):
        count_letter += 1

# Next, we'll count the number of words in the text
    if (text[i].isspace()):
        count_word += 1

# Then, we'll count the number of sentences in the like
    if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
        count_sentence += 1

calculation = (0.0588 * count_letter / count_word * 100) - (0.296 * count_sentence / count_word * 100) - 15.8

index = round(calculation)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")